src/selectors/entities/users.ts

```diff
@@ -384,10 +383,22 @@ export function searchProfiles(state: GlobalState, term: string, skipCurrent = f
     return filteredProfiles;
 }
 
+export function searchProfilesStartingWithTerm(state: GlobalState, term: string, skipCurrent = false, filters?: Filters): Array<UserProfile> {
```

Comment:
Could you turn these into selector factories like `makeSearchProfiles...`? We should be doing that with all selectors that have parameters to avoid cases like the https://github.com/mattermost/mattermost-redux/pull/1257 where they cause a lot of re-renders.

Reply:
yes, we need memoization, Can we consider these changes out of the scope of the PR? these are not. new changes and I like to have them separately filed

Reply:
Done, but i don't think they usually add much value here. Most of these sectors are used in channel provider and user provider. They. invalidate with new search terms.  Nevertheless changed it to be a factory

Reply:
Yeah, I agree that it's kind of unnecessary, but I want to make a habit of using the factories since it will help prevent mistakes like with `getChannel`. Alternatively, if `term`, `skipCurrent`, and `filters` also lived in the redux state, then we wouldn't need to worry about the parameters or the factory in the first place.